katas/content/grovers_search/prefix_oracle/index.md

```diff
@@ -0,0 +1,13 @@
+**Inputs:** 
+
+1. $N$ qubits in an arbitrary state $\ket{x}$ (input/query register).
+2. A qubit in an arbitrary state $\ket{y}$ (output/target qubit).
+3. A bit pattern $p$ of length $P$ represented as a `Bool[]` ($1 ≤ P ≤ N$).
+
+**Goal:** 
+Implement a quantum oracle which checks whether the pattern $p$ is the prefix of the input register, that is, for all $j$ between $0$ and $P - 1$, inclusive, $p_j = x_j$. (`false` and `true` values represent states $\ket{0}$ and $\ket{1}$, respectively).
```

Comment:
The assignment doesn't say what to do with qubit `y` explicitly. I understand that we are writing a marking oracle, but it would be nice to provide complete assignment. For example, phase oracle assignment, which comes later in this Kata, tells to 'flip the phase'.